lib/worker.js

```diff
@@ -41,213 +37,219 @@ module.exports = Worker
+  _eventTags () {
+    const tokens = this.queue.split('.').reverse()
+    let lastToken = ''
+    let tags = tokens.reduce(function (acc, currentValue, currentIndex) {
+      const key = 'token' + currentIndex
+      const newToken = currentIndex === 0
+        ? currentValue
+        : currentValue + '.' + lastToken
+      acc[key] = newToken
+      lastToken = newToken
+      return acc
+    }, {})
+    tags.queue = this.queue
+    return tags
+  }
 
-/**
- * Helper function calling `monitor.increment`.
- * Monitor wouldn't be called if `process.env.WORKER_MONITOR_DISABLED` set.
- * @param {string} eventName name to be reported into the datadog
- * @param {object} [extraTags] extra tags to be send with the event
- * @private
- */
-Worker.prototype._incMonitor = function (eventName, extraTags) {
-  if (process.env.WORKER_MONITOR_DISABLED) {
-    return
+  /**
+   * Helper function calling `monitor.increment`.
+   * Monitor wouldn't be called if `process.env.WORKER_MONITOR_DISABLED` set.
+   * @param {string} eventName name to be reported into the datadog
+   * @param {object} [extraTags] extra tags to be send with the event
+   * @private
+   */
+  _incMonitor (eventName, extraTags) {
+    if (process.env.WORKER_MONITOR_DISABLED) {
+      return
+    }
+    let tags = this._eventTags()
+    if (extraTags) {
+      tags = merge(tags, extraTags)
+    }
+    monitor.increment(eventName, tags)
   }
-  var tags = this._eventTags()
-  if (extraTags) {
-    tags = merge(tags, extraTags)
+
+  /**
+   * Helper function calling `monitor.timer`.
+   * Timer wouldn't be created if `process.env.WORKER_MONITOR_DISABLED` set
+   * @return {object} new timer
+   * @private
+   */
+  _createTimer () {
+    const tags = this._eventTags()
+    return !process.env.WORKER_MONITOR_DISABLED
+      ? monitor.timer('ponos.timer', true, tags)
+      : null
   }
-  monitor.increment(eventName, tags)
-}
 
-/**
- * Helper function calling `monitor.timer`.
- * Timer wouldn't be created if `process.env.WORKER_MONITOR_DISABLED` set
- * @return {object} new timer
- * @private
- */
-Worker.prototype._createTimer = function () {
-  var tags = this._eventTags()
-  return !process.env.WORKER_MONITOR_DISABLED
-    ? monitor.timer('ponos.timer', true, tags)
-    : null
-}
+  /**
+   * Runs the worker. If the task for the job fails, then this method will retry
+   * the task (with an exponential backoff) a number of times defined by the
+   * environment of the process.
+   * @returns {promise} Promise resolved once the task succeeds or fails.
+   */
+  run () {
+    const jobAndQueueData = {
+      queue: this.queue,
+      job: this.job
+    }
+    this._incMonitor('ponos')
+    const timer = this._createTimer()
+    return Promise.resolve().bind(this)
+      .then(function runTheTask () {
+        const attemptData = {
+          attempt: this.attempt++,
+          timeout: this.msTimeout
+        }
+        this.log.info(merge(jobAndQueueData, attemptData), 'Running task')
```

Comment:
We should use `.child` to create a logger child which automatically has the relevant worker data for this instance in the constructor above. By doing so we can completely rid ourselves of the `merge` methods. It will clean things up on logs for this method considerably.